sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/KalixServer.scala

```diff
@@ -231,50 +234,43 @@ class KalixServer(applicationContext: ApplicationContext, config: Config) {
     }
 
   private def actionProvider[A <: Action](clz: Class[A]): ActionProvider[A] =
-    if (hasContextConstructor(clz, classOf[ActionCreationContext]))
-      ReflectiveActionProvider.of(
-        clz,
-        context => {
+    ReflectiveActionProvider.of(
+      clz,
+      context => {
+        if (hasContextConstructor(clz, classOf[ActionCreationContext]))
           threadLocalActionContext.set(context)
-          kalixBeanFactory.getBean(clz)
-        })
-    else
-      ReflectiveActionProvider.of(clz, _ => kalixBeanFactory.getBean(clz))
+
+        val grpcClients = GrpcClients.get(context.materializer().system)
```

Comment:
```suggestion
        val grpcClients = GrpcClients(context.materializer().system)
```

The `.get` is usually for Java APIs.